.github/workflows/release-tag.yml

```diff
@@ -0,0 +1,55 @@
+# if changing this name, also update promotion.yaml
+name: release-tag
+
+on:
+  push:
+    tags:
+      - v*
+
+jobs:
+  golint:
+    name: vet fmt and lint
+    runs-on: ubuntu-20.04
+    container: golang:1.13.7
+    steps:
+    - uses: actions/checkout@v2.3.3
+    - name: Install golint
+      run: go get -u golang.org/x/lint/golint
+    - run: make lint
+      shell: bash
+  gotest:
+    name: go test
+    runs-on: ubuntu-20.04
+    container: golang:1.13.7
+    steps:
+    - uses: actions/checkout@v2.3.3
+    - run: make test
+  docker_build:
+    name: docker_build
+    runs-on: ubuntu-20.04
+    container:
+      image: docker:20.10.5
+      options: -t
+    # Setting up dind service container
+    services:
+      docker:
+        image: docker:20.10.5-dind
+        env:
+          DOCKER_DRIVER: overlay
+          DOCKER_HOST: tcp://localhost:2375
+    steps:
+    - name: Install Tools
+      run: apk add --update make git jq rsync curl
+    - uses: actions/checkout@v2.3.3
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v1
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+    - name: Login to registry
+      uses: docker/login-action@v1
+      with:
+        registry: quay.io
+        username: ${{ secrets.QUAY_USER }}
+        password: ${{ secrets.QUAY_PASSWORD }}
+    - name: Build images
+      run: make push-docker-image
```

Comment:
Shame that all but the last step is duplicated here. I wonder if we could have another task which pulls the image for the sha of the tag. Then tags that as the release, and pushes to quay?

Reply:
I have tried this:

- pull this image which is multi-platform: 
![image](https://user-images.githubusercontent.com/10594577/114579703-68067c80-9c7e-11eb-9a1f-6a9ca9a57b4a.png)
```
$ docker pull josefuentes/kubectl:v1.20.0
v1.20.0: Pulling from josefuentes/kubectl
75646c2fb410: Already exists 
97150b587ac2: Pull complete 
5a321e5a7e17: Pull complete 
29f06f9cf893: Pull complete 
Digest: sha256:2daebe4eae191d4ebc2c58a600c988d9f43571d1a972cae839a806d47c31a1ad
Status: Downloaded newer image for josefuentes/kubectl:v1.20.0
docker.io/josefuentes/kubectl:v1.20.0
```

- Tag it with another tag:

```
$ docker tag josefuentes/kubectl:v1.20.0 josefuentes/kubectl:v1.20.0-copy

```


- Push the new tag:
```
$ docker push josefuentes/kubectl:v1.20.0-copy                           
The push refers to repository [docker.io/josefuentes/kubectl]
c317b8b2eb88: Layer already exists 
759b01e303b5: Layer already exists 
cf32df1a228a: Layer already exists 
346fddbbb0ff: Layer already exists 
v1.20.0-copy: digest: sha256:da9fa76c7a3761d63c46436e2a1ab44f2880518f4531dc1f5b64d99cc0a2209f size: 1160
```



As you can see, the resulting image is not multiplatform:

![image](https://user-images.githubusercontent.com/10594577/114579967-b0be3580-9c7e-11eb-9b50-12a2da9131cf.png)



If we want to do that, we need to find some `buildx` magic. I don't see a clean simple solution at the moment.

The best I can think of implies using another Dockerfile to build a new image just with `FROM <image>:<commit>`.

But I don't think we want that, The sha of the resulting image will be different than the original one anyway.

Reply:
Nevermind! I have found a way!

Reply:
this should do the job: d836c1f

I have tested the command locally and ended up with this:

```
docker buildx imagetools create josefuentes/kubectl:v1.20.0 --tag josefuentes/kubectl:v1.20.0-copy2       
docker.io/josefuentes/kubectl:v1.20.0-copy2
```


![image](https://user-images.githubusercontent.com/10594577/114583472-e7e21600-9c81-11eb-9b91-13eb7205413b.png)

Reply:
Looks nice!